katdal/visdatav4.py

```diff
@@ -103,43 +107,28 @@ def __init__(self, source, ref_ant='', time_offset=0.0, centre_freq=None,
         self.source = source
         self.file = {}
         self.version = '4.0'
-        stream_name = 'sdp_l0'
-        self.dump_period = attrs[stream_name + '_int_time']
-        self._timestamps = source.timestamps[:]
-        self._keepdims = keepdims
-
-        # Check dimensions of timestamps vs those of visibility data
-        num_dumps = len(self._timestamps)
-        if source.data and (num_dumps != source.data.shape[0]):
-            raise BrokenFile('Number of timestamps received from ingest '
-                             '(%d) differs from number of dumps in data (%d)' %
-                             (num_dumps, source.data.shape[0]))
-        # The expected_dumps should always be an integer (like num_dumps),
-        # unless the timestamps and/or dump period are messed up in the file,
-        # so threshold of this test is a bit arbitrary (e.g. could use > 0.5).
-        # The last dump might only be partially filled by ingest, so ignore it.
-        if num_dumps > 1:
-            expected_dumps = 2 + (self._timestamps[-2] -
-                                  self._timestamps[0]) / self.dump_period
-            if abs(expected_dumps - num_dumps) >= 0.01:
-                # Warn the user, as this is anomalous
-                logger.warning("Irregular timestamps detected: expected %.3f "
-                               "dumps based on dump period and start/end times, "
-                               "got %d instead", expected_dumps, num_dumps)
-        self._timestamps += self.time_offset
-        if self._timestamps[0] < 1e9:
+        self.dump_period = attrs['int_time']
+        num_dumps = len(source.timestamps)
+        source.timestamps += self.time_offset
+        if source.timestamps[0] < 1e9:
             logger.warning("Data set has invalid first correlator timestamp "
-                           "(%f)", self._timestamps[0])
+                           "(%f)", source.timestamps[0])
         half_dump = 0.5 * self.dump_period
-        self.start_time = katpoint.Timestamp(self._timestamps[0] - half_dump)
-        self.end_time = katpoint.Timestamp(self._timestamps[-1] + half_dump)
-        self._time_keep = np.ones(num_dumps, dtype=np.bool)
+        self.start_time = katpoint.Timestamp(source.timestamps[0] - half_dump)
+        self.end_time = katpoint.Timestamp(source.timestamps[-1] + half_dump)
+        self._time_keep = np.full(num_dumps, True)
+        all_dumps = [0, num_dumps]
 
         # Assemble sensor cache
-        self.sensor = SensorCache(source.metadata.sensors, self._timestamps,
+        self.sensor = SensorCache(source.metadata.sensors, source.timestamps,
                                   self.dump_period, self._time_keep,
                                   SENSOR_PROPS, VIRTUAL_SENSORS, SENSOR_ALIASES)
 
+        # ------ Extract flags ------
+
+        self._flags_select = np.array([0], dtype=np.uint8)
```

Comment:
Isn't this going to be overwritten by the setter invoked on the next line?

Reply:
Sure - it's a "default default" that reminds me that such a variable exists, otherwise it magically appears later on.

Reply:
Fair enough. Reading through the constructor it looks like this is the value it has at the end of construction though - a comment might help.

Reply:
Commented and changed to 255 instead (the intended initial value).